common/util/file_util.cc

```diff
@@ -46,6 +48,46 @@ absl::string_view Stem(absl::string_view filename) {
              : filename.substr(0, last_dot_pos);
 }
 
+// TODO (after bump to c++17) rewrite this function to use std::filesystem
+bool UpwardFileSearch(absl::string_view start, absl::string_view filename,
+                      std::string* result) {
+  static const std::string dir_separator = "/";
+
+  /* Convert to absolute path */
+  char absolute_path[PATH_MAX];
+  if (realpath(start.data(), absolute_path) < 0) {
+    return false;
+  }
+
+  /* Add trailing slash */
+  std::string init = absl::StrCat(absolute_path, dir_separator);
+
+  VLOG(1) << "Upward search for " << filename << ", starting in " << start;
+
+  size_t r = 0, n = init.length();
```

Comment:
I'd name `n` a bit more so that the code below reads more fluently without having to wonder what `n` means.
Maybe something like `search_up_to` ?